reana-ui/src/pages/workflowList/components/WorkflowStatusFilter.js

```diff
@@ -2,39 +2,80 @@
   -*- coding: utf-8 -*-
 
   This file is part of REANA.
-  Copyright (C) 2020 CERN.
+  Copyright (C) 2020, 2022 CERN.
 
   REANA is free software; you can redistribute it and/or modify it
   under the terms of the MIT License; see LICENSE file for more details.
 */
 
+import isEqual from "lodash/isEqual";
 import PropTypes from "prop-types";
-import { Dropdown } from "semantic-ui-react";
+import { useEffect, useState } from "react";
+import { Checkbox, Dropdown, Grid } from "semantic-ui-react";
 
+import { WORKFLOW_STATUSES } from "~/config";
 import { statusMapping } from "~/util";
 
-const statusOptions = Object.keys(statusMapping).map((status) => ({
+function removeDeleted(statuses) {
+  return statuses.filter((status) => status !== "deleted");
+}
+
+const NON_DELETED_STATUSES = removeDeleted(WORKFLOW_STATUSES);
+
+const statusOptions = WORKFLOW_STATUSES.map((status) => ({
   key: status,
   text: status,
   value: status,
   icon: statusMapping[status].icon,
 }));
 
-export default function WorkflowStatusFilters({ valueList, filter }) {
+const nonDeletedStatusOptions = statusOptions.filter(({ value: status }) =>
+  NON_DELETED_STATUSES.includes(status)
+);
+
+export default function WorkflowStatusFilters({ statusFilter, filter }) {
+  const [valueList, setValueList] = useState([]);
+  const [showDeleted, setShowDeleted] = useState(false);
```

Comment:
Small issue appeared when you changed to hide workflows by default. It triggers two API requests now - one to fetch all workflow statuses and then after `useEffect` triggers it filters all except deleted. I think we should set `NON_DELETED_STATUSES` by default [here](https://github.com/reanahub/reana-ui/blob/master/reana-ui/src/pages/workflowList/WorkflowList.js#L51) to solve it